lib/azure/armrest/model/base_model.rb

```diff
@@ -33,10 +33,21 @@ class BaseModel < Delegator
       #
       def initialize(json)
         @json = json
+        @resource_group = nil
         @ostruct = JSON.parse(json, object_class: OpenStruct)
         __setobj__(@ostruct)
       end
 
+      # Return the resource group for the current object.
+      def resource_group
+        @resource_group ||= id[/resourceGroups\/(.+?)\//i, 1] rescue nil
+      end
+
+      # Return a hash of tags associated with the resource.
+      def tags
```

Comment:
I ran some tests. `tags` is optional. So you have to deal with the case that `tags` is `nil`.
Besides, let's make this declarative so any child class can easily add attributes that need to be remain in Hash format, for example `parameters` for deployments. I would prefer a syntax like this
`hash_property :tags, :parameters`